packages/ai-core/src/artifact-api.ts

```diff
@@ -23,68 +23,86 @@ export const ArtifactApi = {
    * Search by substring of artifact name or description, if required.
    *
    * @param queryParameters - Object containing the following keys: scenarioId, executionId, name, kind, artifactLabelSelector, $top, $skip, $search, searchCaseInsensitive, $expand.
+   * @param headerParameters - Object containing the following keys: AI-Resource-Group.
    * @returns The request builder, use the `execute()` method to trigger the request.
    */
-  artifactQuery: (queryParameters?: {
-    scenarioId?: string;
-    executionId?: string;
-    name?: Name_1;
-    kind?: 'model' | 'dataset' | 'resultset' | 'other';
-    artifactLabelSelector?: string[];
-    $top?: number;
-    $skip?: number;
-    $search?: string;
-    searchCaseInsensitive?: boolean;
-    $expand?: 'scenario';
-  }) =>
+  artifactQuery: (
```

Comment:
[q] Why is `queryParameters` no longer optional? (Valid for all the API introduced in this PR.)

Reply:
Support for generating OpenAPI clients with header parameters was added in [v3.16.0](https://github.com/SAP/cloud-sdk-js/releases/tag/v3.16.0). Here's the related [PR](https://github.com/SAP/cloud-sdk-js/pull/4708) if you would like the context. The main reason is since headerParameters is required, an optional parameter (queryParameters) cannot come before it. So even though the properties in queryParams are optional, the object itself has to passed (could be empty {}).